In [1]:
import pandas as pd
import pandas as pd
from sqlalchemy import create_engine
%load_ext sql

In [2]:
# Create a connection to the PostgreSQL database
engine=create_engine('postgresql+psycopg2://postgres:password@localhost:5432/medsupply')
connection=engine.connect()
print("Connected:", connection.closed == False)

Connected: True


In [3]:
%sql postgresql+psycopg2://postgres:password@localhost:5432/medsupply

'Connected: postgres@medsupply'

### Percentage Revenue for each Invoice status

In [4]:

%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Completed' THEN tm.grand_total
            ELSE 0 
            END)
            AS completed_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    completed_revenue,
    total_revenue,
    ROUND((completed_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS completed_revenue_percentage
FROM revenue_cte;


 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(657245355638.0, 1496017486673.0, Decimal('43.93'))]

dari total revenue 1115631508288, hampir 56.22% invoice yang sudah bisa di cairkan (complete) --> versi 1

dari total revenue 	1496017486673, hanya 43.93% invoice yang sudah bisa di cairkan (complete) dan itu kurang dari setengahnya --> versi 2

In [6]:

%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Sent to Tax' THEN tm.grand_total
            ELSE 0 
            END)
            AS sent_to_tax_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    sent_to_tax_revenue,
    total_revenue,
    ROUND((sent_to_tax_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS sent_to_tax_revenue_percentage
FROM revenue_cte;


 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(269415855721.0, 1496017486673.0, Decimal('18.01'))]

sementara ada 11.19% yang masih dalam proses naik pajak dan belum di lakukan tukar faktur/pencairan dana dengan customer --> versi 1

sementara ada 18.01% yang masih dalam proses naik pajak dan belum di lakukan tukar faktur/pencairan dana dengan customer --> versi 2

In [8]:
%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Awaiting DN Signed' THEN tm.grand_total
            ELSE 0 
            END)
            AS awaiting_dn_signed_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    awaiting_dn_signed_revenue,
    total_revenue,
    ROUND((awaiting_dn_signed_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS awaiting_dn_signed_revenue_percentage
FROM revenue_cte;

 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(195101560399.0, 1496017486673.0, Decimal('13.04'))]

ada hampir 17% transaksi yang belum bisa naik pajak karena menunggu DN dari rumah sakit/klinik setelah barang di kirim --> versi 1

ada 13.04% transaksi yang belum bisa naik pajak karena menunggu DN dari rumah sakit/klinik setelah barang di kirim --> versi 2

In [9]:
%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Customer Confirmation' THEN tm.grand_total
            ELSE 0 
            END)
            AS customer_confirmation_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    customer_confirmation_revenue,
    total_revenue,
    ROUND((customer_confirmation_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS customer_confirmation_revenue_percentage
FROM revenue_cte;

 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(300021056099.0, 1496017486673.0, Decimal('20.05'))]

ada transaksi yang menunggu confirmasi dari customer karena barang dari PO customer ada sebagian inden sebanyak 9.67% --> versi 1

ada transaksi yang menunggu confirmasi dari customer karena barang dari PO customer ada sebagian inden lumayan besar yaitu 20.05% --> versi 2

In [10]:
%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status IS NULL THEN tm.grand_total
            ELSE 0 
            END)
            AS completed_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    completed_revenue,
    total_revenue,
    ROUND((completed_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS completed_revenue_percentage
FROM revenue_cte;

 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(27298304418.0, 1496017486673.0, Decimal('1.82'))]

4.8% transaksi masih menggantung statusnya dan ini lumayan besar dan harus di telusuri --> versi 1

1.82% transaksi masih menggantung statusnya, tidak terlalu tinggi tapi tetap harus di telusuri --> versi 2

In [11]:
%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Cancelled' THEN tm.grand_total
            ELSE 0 
            END)
            AS completed_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    completed_revenue,
    total_revenue,
    ROUND((completed_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS completed_revenue_percentage
FROM revenue_cte;

 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(46935354398.0, 1496017486673.0, Decimal('3.14'))]

kurang dari 2% setelah confirmasi dari customer, transaksi di cancel terkait barang yang inden --> versi 1

ada 3.14% transaksi yang di cancel oleh customer, ini cukup tinggi --> versi 2